### DISCUSSION TITLE
about Nue UI & Server components

### DISCUSSION THREAD

**Discussion by wangzengzhi** [nuejs/nue#231] [General]
Are there any plans for Nue UI? Will the server-side components be supported in the future?

**Comment by tipiirai** [comment]
Yes! Definite plans for Nue UI. Nue CSS is released next week, which will shed some light on the future of Nue in terms of styling, design systems, UI libraries, and templates. 

Server-side components will always be supported, no question about it. What made you concerned about it?

**Reply by wangzengzhi** [reply]
Thank you. The reason is that the content related to server-side components cannot be seen on the official website.

**Reply by tipiirai** [reply]
Ah. Good point. Needs to be added there obviously. Thanks!

**Reply by itsjavi** [reply]
Full-stack use cases and tutorials would be appreciated, specially examples accessing DB / session and feeding that to the frontend. That's the missing piece.

**Reply by aaronpeacock** [reply]
I don't see any material yet about integrating Nue JS built frontends into web pages served by an existing backend of some kind.

Generally, I found nothing in the docs explaining any server functionality (which one may want or wish to avoid, aka Svelte vs Sveltekit) 

I like the radical simplicity of the root site.yaml config, and the `dist` setting.
Am I to understand that *any* directory structure will be preserved in the build directory when doing `nue build`?

I am just trying to understand how to apply nue to my current reality:
- a server rendered template with server-data embedded (different server-routes going to different pages)
-  a javascript bundle loaded via a script tag on this same server-rendered page
-  the javascript can do api calls to the same server

**Reply by aaronpeacock** [reply]
I guess Nue is more for making either an SPA or a static site, or a mix of these 2,  rather than making bundles to load on individual pages?

**Reply by tipiirai** [reply]
Backend & database integration is indeed missing from the docs. Nue is still at early stages with many basic stuff still missing: most critically a UI library, full-stack dev environment, and quick starter templates for blogs, websites, and SPA's. 

@aaronpeacock I suggest you join our Slack channel for more imediate answers. Link is found on top/right corner of nuejs.org
